### ISSUE TITLE
F750 CU 3x400V

### ISSUE THREAD

**Issue by allindigital** [hp-net/homebridge-nibe#14]
**What sensors/parameters should be supported**
- Outside Temperature
- Hot Water Temperature

**Data from logs**
```
{
  "unitData": [
    {
      "systemUnitId": 0,
      "name": "master",
      "shortName": "m",
      "product": "F750 CU 3x400V",
      "softwareVersion": "9086R10",
      "categories": [
        {
          "categoryId": "STATUS",
          "name": "status",
          "parameters": [
            {
              "parameterId": 40067,
              "name": "40067",
              "title": "avg. outdoor temp",
              "designation": "BT1",
              "unit": "°C",
              "displayValue": "4.0°C",
              "rawValue": 40,
              "key": "AVG_OUTDOOR_TEMP_BT1",
              "divideBy": 10,
              "value": 4
            },
            {
              "parameterId": 40014,
              "name": "40014",
              "title": "hot water charging",
              "designation": "BT6",
              "unit": "°C",
              "displayValue": "47.7°C",
              "rawValue": 477,
              "key": "HOT_WATER_CHARGING_BT6",
              "divideBy": 10,
              "value": 47.7
            },
            {
              "parameterId": 40013,
              "name": "40013",
              "title": "hot water top",
              "designation": "BT7",
              "unit": "°C",
              "displayValue": "49.5°C",
              "rawValue": 495,
              "key": "HOT_WATER_TOP_BT7",
              "divideBy": 10,
              "value": 49.5
            },
            {
              "parameterId": 40004,
              "name": "40004",
              "title": "outdoor temp.",
              "designation": "BT1",
              "unit": "°C",
              "displayValue": "8.5°C",
              "rawValue": 85,
              "key": "OUTDOOR_TEMP_BT1",
              "divideBy": 10,
              "value": 8.5
            },
            {
              "parameterId": 40083,
              "name": "40083",
              "title": "current",
              "designation": "EB100-BE1",
              "unit": "A",
              "displayValue": "0.8A",
              "rawValue": 8,
              "key": "CURRENT_EB100_BE1",
              "divideBy": 10,
              "value": 0.8
            },
            {
              "parameterId": 40081,
              "name": "40081",
              "title": "current",
              "designation": "EB100-BE2",
              "unit": "A",
              "displayValue": "0.8A",
              "rawValue": 8,
              "key": "CURRENT_EB100_BE2",
              "divideBy": 10,
              "value": 0.8
            },
            {
              "parameterId": 40079,
              "name": "40079",
              "title": "current",
              "designation": "EB100-BE3",
              "unit": "A",
              "displayValue": "0.3A",
              "rawValue": 3,
              "key": "CURRENT_EB100_BE3",
              "divideBy": 10,
              "value": 0.3
            },
            {
              "parameterId": 43005,
              "name": "43005",
              "title": "degree minutes",
              "designation": "",
              "unit": "DM",
              "displayValue": "89DM",
              "rawValue": 894,
              "key": "DEGREE_MINUTES",
              "divideBy": 10,
              "value": 89.4
            }
          ]
        },
        {
          "categoryId": "CPR_INFO_EP14",
          "name": "compressor module ",
          "parameters": [
            {
              "parameterId": 10012,
              "name": "10012",
              "title": "blocked",
              "designation": "",
              "unit": "",
              "displayValue": "no",
              "rawValue": 0,
              "key": "BLOCKED",
              "divideBy": 0,
              "value": 0
            },
            {
              "parameterId": 43416,
              "name": "43416",
              "title": "compressor starts",
              "designation": "EB100-EP14",
              "unit": "",
              "displayValue": "18357",
              "rawValue": 18357,
              "key": "COMPRESSOR_STARTS_EB100_EP14",
              "divideBy": 0,
              "value": 18357
            },
            {
              "parameterId": 43372,
              "name": "43372",
              "title": "evaporating limit",
              "designation": "",
              "unit": "",
              "displayValue": "0",
              "rawValue": 0,
              "key": "EVAPORATING_LIMIT",
              "value": 0
            },
            {
              "parameterId": 43064,
              "name": "43064",
              "title": "heat: dT calc. value",
              "designation": "",
              "unit": "",
              "displayValue": "--",
              "rawValue": -32768,
              "key": "HEAT_DT_CALC_VALUE",
              "divideBy": 10,
              "value": -3276.8
            },
            {
              "parameterId": 43065,
              "name": "43065",
              "title": "heat: dT is (BT12 | BT63)-BT3",
              "designation": "",
              "unit": "",
              "displayValue": "--",
              "rawValue": -32768,
              "key": "HEAT_DT_IS_BT12_BT63_BT3",
              "divideBy": 10,
              "value": -3276.8
            },
            {
              "parameterId": 43371,
              "name": "43371",
              "title": "hotgas limit",
              "designation": "",
              "unit": "",
              "displayValue": "0",
              "rawValue": 0,
              "key": "HOTGAS_LIMIT",
              "value": 0
            },
            {
              "parameterId": 43181,
              "name": "43181",
              "title": "speed heat med. pump 1",
              "designation": "",
              "unit": "%",
              "displayValue": "0%",
              "rawValue": 0,
              "key": "SPEED_HEAT_MED_PUMP_1",
              "value": 0
            },
            {
              "parameterId": 40017,
              "name": "40017",
              "title": "condenser out",
              "designation": "EB100-EP14-BT12",
              "unit": "°C",
              "displayValue": "32.0°C",
              "rawValue": 320,
              "key": "CONDENSER_OUT_EB100_EP14_BT12",
              "divideBy": 10,
              "value": 32
            },
            {
              "parameterId": 40012,
              "name": "40012",
              "title": "condenser return",
              "designation": "EB100-EP14-BT3",
              "unit": "°C",
              "displayValue": "32.9°C",
              "rawValue": 329,
              "key": "RETURN_TEMP_EB100_EP14_BT3",
              "divideBy": 10,
              "value": 32.9
            },
            {
              "parameterId": 43146,
              "name": "43146",
              "title": "dT Inverter - exhaust air",
              "designation": "BT20",
              "unit": "°C",
              "displayValue": "3.0°C",
              "rawValue": 30,
              "key": "DT_INVERTER_EXHAUST_AIR",
              "divideBy": 10,
              "value": 3
            },
            {
              "parameterId": 40020,
              "name": "40020",
              "title": "evaporator",
              "designation": "EB100-BT16",
              "unit": "°C",
              "displayValue": "23.3°C",
              "rawValue": 233,
              "key": "EVAPORATOR_EB100_BT16",
              "divideBy": 10,
              "value": 23.3
            },
            {
              "parameterId": 40018,
              "name": "40018",
              "title": "hot gas",
              "designation": "EB100-EP14-BT14",
              "unit": "°C",
              "displayValue": "39.0°C",
              "rawValue": 390,
              "key": "HOT_GAS_EB100_EP14_BT14",
              "divideBy": 10,
              "value": 39
            },
            {
              "parameterId": 43115,
              "name": "43115",
              "title": "HW: charge calc value",
              "designation": "",
              "unit": "°C",
              "displayValue": "--",
              "rawValue": -32768,
              "key": "HW_CHARGE_CALC_VALUE",
              "divideBy": 10,
              "value": -3276.8
            },
            {
              "parameterId": 43116,
              "name": "43116",
              "title": "HW:curr charge val BT12|BT63",
              "designation": "",
              "unit": "°C",
              "displayValue": "31.4°C",
              "rawValue": 314,
              "key": "HW_CURR_CHARGE_VAL_BT12_BT63",
              "divideBy": 10,
              "value": 31.4
            },
            {
              "parameterId": 40019,
              "name": "40019",
              "title": "liquid line",
              "designation": "EB100-EP14-BT15",
              "unit": "°C",
              "displayValue": "33.0°C",
              "rawValue": 330,
              "key": "LIQUID_LINE_EB100_EP14_BT15",
              "divideBy": 10,
              "value": 33
            },
            {
              "parameterId": 40022,
              "name": "40022",
              "title": "suction gas",
              "designation": "EB100-EP14-BT17",
              "unit": "°C",
              "displayValue": "24.2°C",
              "rawValue": 242,
              "key": "SUCTION_GAS_EB100_EP14_BT17",
              "divideBy": 10,
              "value": 24.2
            },
            {
              "parameterId": 43420,
              "name": "43420",
              "title": "compressor operating time",
              "designation": "EB100-EP14",
              "unit": "h",
              "displayValue": "32142h",
              "rawValue": 32142,
              "key": "COMPRESSOR_OPERATING_TIME_EB100_EP14",
              "divideBy": 0,
              "value": 32142
            },
            {
              "parameterId": 43424,
              "name": "43424",
              "title": "compressor operating time hot water",
              "designation": "EB100-EP14",
              "unit": "h",
              "displayValue": "7862h",
              "rawValue": 7862,
              "key": "COMPRESSOR_OPERATING_TIME_HOT_WATER_EB100_EP14",
              "divideBy": 0,
              "value": 7862
            },
            {
              "parameterId": 43123,
              "name": "43123",
              "title": "allowed compr. freq.",
              "designation": "",
              "unit": "Hz",
              "displayValue": "60Hz",
              "rawValue": 60,
              "key": "ALLOWED_COMPR_FREQ",
              "value": 60
            },
            {
              "parameterId": 43136,
              "name": "43136",
              "title": "current compr. frequency",
              "designation": "",
              "unit": "Hz",
              "displayValue": "0Hz",
              "rawValue": 0,
              "key": "CURRENT_COMPR_FREQUENCY",
              "divideBy": 10,
              "value": 0
            },
            {
              "parameterId": 43144,
              "name": "43144",
              "title": "compressor time factor",
              "designation": "",
              "unit": "kWh",
              "displayValue": "20477.6kWh",
              "rawValue": 2047756,
              "key": "COMPRESSOR_TIME_FACTOR",
              "divideBy": 100,
              "value": 20477.56
            },
            {
              "parameterId": 43305,
              "name": "43305",
              "title": "compressor time factor hot water",
              "designation": "",
              "unit": "kWh",
              "displayValue": "6527.7kWh",
              "rawValue": 652765,
              "key": "COMPRESSOR_TIME_FACTOR_HOT_WATER",
              "divideBy": 100,
              "value": 6527.65
            }
          ]
        },
        {
          "categoryId": "VENTILATION",
          "name": "ventilation",
          "parameters": [
            {
              "parameterId": 10001,
              "name": "10001",
              "title": "fan speed",
              "designation": "",
              "unit": "%",
              "displayValue": "50%",
              "rawValue": 50,
              "key": "FAN_SPEED",
              "divideBy": 0,
              "value": 50
            },
            {
              "parameterId": 40025,
              "name": "40025",
              "title": "exhaust air",
              "designation": "BT20",
              "unit": "°C",
              "displayValue": "24.1°C",
              "rawValue": 241,
              "key": "EXHAUST_AIR_BT20",
              "divideBy": 10,
              "value": 24.1
            },
            {
              "parameterId": 40026,
              "name": "40026",
              "title": "extract air",
              "designation": "BT21",
              "unit": "°C",
              "displayValue": "23.6°C",
              "rawValue": 236,
              "key": "EXTRACT_AIR_BT21",
              "divideBy": 10,
              "value": 23.6
            }
          ]
        },
        {
          "categoryId": "DEFROSTING",
          "name": "defrosting",
          "parameters": [
            {
              "parameterId": 43124,
              "name": "43124",
              "title": "reference, air velocity sensor",
              "designation": "",
              "unit": "",
              "displayValue": "--",
              "rawValue": -32768,
              "key": "REFERENCE_AIR_VELOCITY_SENSOR",
              "divideBy": 10,
              "value": -3276.8
            },
            {
              "parameterId": 40050,
              "name": "40050",
              "title": "value, air velocity sensor",
              "designation": "EB100-BS1",
              "unit": "",
              "displayValue": "115.5",
              "rawValue": 1155,
              "key": "VALUE_AIR_VELOCITY_SENSOR_EB100_BS1",
              "divideBy": 10,
              "value": 115.5
            },
            {
              "parameterId": 43125,
              "name": "43125",
              "title": "decrease from reference",
              "designation": "",
              "unit": "%",
              "displayValue": "-0.9%",
              "rawValue": -9,
              "key": "DECREASE_FROM_REFERENCE",
              "divideBy": 10,
              "value": -0.9
            },
            {
              "parameterId": 43066,
              "name": "43066",
              "title": "defrosting time",
              "designation": "",
              "unit": "s",
              "displayValue": "0s",
              "rawValue": 0,
              "key": "DEFROSTING_TIME",
              "value": 0
            }
          ]
        },
        {
          "categoryId": "SYSTEM_1",
          "name": "climate system 1",
          "parameters": [
            {
              "parameterId": 43161,
              "name": "43161",
              "title": "external adjustment",
              "designation": "S1",
              "unit": "",
              "displayValue": "no",
              "rawValue": 0,
              "key": "EXTERNAL_ADJUSTMENT_S1",
              "divideBy": 0,
              "value": 0
            },
            {
              "parameterId": 43009,
              "name": "43009",
              "title": "calculated flow temp.",
              "designation": "S1",
              "unit": "°C",
              "displayValue": "26.8°C",
              "rawValue": 268,
              "key": "CALCULATED_FLOW_TEMP_S1",
              "divideBy": 10,
              "value": 26.8
            },
            {
              "parameterId": 40047,
              "name": "40047",
              "title": "heat medium flow",
              "designation": "1 rad EB100-BT61",
              "unit": "°C",
              "displayValue": "26.6°C",
              "rawValue": 266,
              "key": "HEAT_MEDIUM_FLOW_EB100_BT61",
              "divideBy": 10,
              "value": 26.6
            },
            {
              "parameterId": 40008,
              "name": "40008",
              "title": "heat medium flow",
              "designation": "EB100-BT63",
              "unit": "°C",
              "displayValue": "31.7°C",
              "rawValue": 317,
              "key": "HEAT_MEDIUM_FLOW_BT2",
              "divideBy": 10,
              "value": 31.7
            },
            {
              "parameterId": 40048,
              "name": "40048",
              "title": "heat medium return",
              "designation": "1 rad EB100-BT62",
              "unit": "°C",
              "displayValue": "26.6°C",
              "rawValue": 266,
              "key": "HEAT_MEDIUM_RETURN_EB100_BT62",
              "divideBy": 10,
              "value": 26.6
            },
            {
              "parameterId": 40033,
              "name": "40033",
              "title": "room temperature",
              "designation": "BT50",
              "unit": "°C",
              "displayValue": "23.9°C",
              "rawValue": 239,
              "key": "ROOM_TEMPERATURE_BT50",
              "divideBy": 10,
              "value": 23.9
            }
          ]
        },
        {
          "categoryId": "ADDITION",
          "name": "addition",
          "parameters": [
            {
              "parameterId": 10033,
              "name": "10033",
              "title": "blocked",
              "designation": "",
              "unit": "",
              "displayValue": "no",
              "rawValue": 0,
              "key": "ADDITION_BLOCKED",
              "divideBy": 0,
              "value": 0
            },
            {
              "parameterId": 47214,
              "name": "47214",
              "title": "fuse size",
              "designation": "",
              "unit": "A",
              "displayValue": "16A",
              "rawValue": 16,
              "key": "ADDITION_FUSE_SIZE",
              "divideBy": 0,
              "value": 16
            },
            {
              "parameterId": 43081,
              "name": "43081",
              "title": "time factor",
              "designation": "",
              "unit": "h",
              "displayValue": "1121.4h",
              "rawValue": 11214,
              "key": "ADDITION_TIME_FACTOR",
              "divideBy": 10,
              "value": 1121.4
            },
            {
              "parameterId": 43084,
              "name": "43084",
              "title": "electrical addition power",
              "designation": "",
              "unit": "kW",
              "displayValue": "0.0kW",
              "rawValue": 0,
              "key": "ELECTRICAL_ADDITION_POWER",
              "divideBy": 10,
              "value": 0
            },
            {
              "parameterId": 47212,
              "name": "47212",
              "title": "set max electrical add.",
              "designation": "",
              "unit": "kW",
              "displayValue": "6.5kW",
              "rawValue": 650,
              "key": "SET_MAX_ELECTRICAL_ADDITION",
              "divideBy": 100,
              "value": 6.5
            }
          ]
        },
        {
          "categoryId": "AUX_IN_OUT",
          "name": "soft in/outputs",
          "parameters": [
            {
              "parameterId": 47411,
              "name": "47411",
              "title": "AUX1",
              "designation": "",
              "unit": "",
              "displayValue": "not used",
              "rawValue": 0,
              "key": "AUX_1",
              "divideBy": 0,
              "value": 0
            },
            {
              "parameterId": 47410,
              "name": "47410",
              "title": "AUX2",
              "designation": "",
              "unit": "",
              "displayValue": "not used",
              "rawValue": 0,
              "key": "AUX_2",
              "divideBy": 0,
              "value": 0
            },
            {
              "parameterId": 47409,
              "name": "47409",
              "title": "AUX3",
              "designation": "",
              "unit": "",
              "displayValue": "not used",
              "rawValue": 0,
              "key": "AUX_3",
              "divideBy": 0,
              "value": 0
            },
            {
              "parameterId": 47408,
              "name": "47408",
              "title": "AUX4",
              "designation": "",
              "unit": "",
              "displayValue": "not used",
              "rawValue": 0,
              "key": "AUX_4",
              "divideBy": 0,
              "value": 0
            },
            {
              "parameterId": 47407,
              "name": "47407",
              "title": "AUX5",
              "designation": "",
              "unit": "",
              "displayValue": "not used",
              "rawValue": 0,
              "key": "AUX_5",
              "divideBy": 0,
              "value": 0
            },
            {
              "parameterId": 47412,
              "name": "47412",
              "title": "X7",
              "designation": "",
              "unit": "",
              "displayValue": "alarm output",
              "rawValue": 0,
              "key": "AUX_X7",
              "divideBy": 0,
              "value": 0
            }
          ]
        },
        {
          "categoryId": "SYSTEM_INFO",
          "name": "info",
          "parameters": [
            {
              "parameterId": 48745,
              "name": "48745",
              "title": "country",
              "designation": "",
              "unit": "",
              "displayValue": "Sweden",
              "rawValue": 45,
              "key": "COUNTRY",
              "divideBy": 0,
              "value": 45
            },
            {
              "parameterId": 0,
              "name": null,
              "title": "product",
              "designation": "",
              "unit": "",
              "displayValue": "F750 CU 3x400V",
              "rawValue": -32768,
              "key": "PRODUCT",
              "value": -32768
            },
            {
              "parameterId": 0,
              "name": null,
              "title": "serial number",
              "designation": "",
              "unit": "",
              "displayValue": "06603615321011",
              "rawValue": -32768,
              "key": "SERIAL_NUMBER",
              "value": -32768
            },
            {
              "parameterId": 0,
              "name": null,
              "title": "version",
              "designation": "",
              "unit": "",
              "displayValue": "9086R10",
              "rawValue": -32768,
              "key": "VERSION",
              "value": -32768
            }
          ]
        }
      ]
    }
  ],
  "manageData": [
    {
      "unit": "",
      "parameters": [
        {
          "parameterId": 48132,
          "name": "48132",
          "title": "Temporary Lux",
          "designation": "",
          "unit": "",
          "displayValue": "off",
          "rawValue": 0,
          "key": "TEMPORARY_LUX",
          "value": 0
        },
        {
          "parameterId": 43427,
          "name": "43427",
          "title": "status",
          "designation": "EP14",
          "unit": "",
          "displayValue": "20.0",
          "rawValue": 20,
          "key": "STATUS_EP14",
          "value": 20
        },
        {
          "parameterId": 43115,
          "name": "43115",
          "title": "HW: charge calc value",
          "designation": "",
          "unit": "°C",
          "displayValue": "--",
          "rawValue": -32768,
          "key": "HW_CHARGE_CALC_VALUE",
          "divideBy": 10,
          "value": -3276.8
        },
        {
          "parameterId": 43064,
          "name": "43064",
          "title": "heat: dT calc. value",
          "designation": "",
          "unit": "",
          "displayValue": "--",
          "rawValue": -32768,
          "key": "HEAT_DT_CALC_VALUE",
          "divideBy": 10,
          "value": -3276.8
        },
        {
          "parameterId": 47260,
          "name": "47260",
          "title": "Selected Fan speed",
          "designation": "",
          "unit": "",
          "displayValue": "normal (50%)",
          "rawValue": 0,
          "key": "SELECTED_FAN_SPEED",
          "value": 0
        }
      ]
    }
  ]
}
```

**Comment by hpruszyn** [comment]
@allindigital please update plugin to v0.0.13 and let me know if accessories from nibe works.

Currently supported: 
* outdoor temperature
* hot water

**Comment by allindigital** [comment]
Hi Hubert.

It is working very well so big thanks for that!
Thank you for your effort!

I will provide screenshots of how it looks from the Nice app and from HomeKit and your integration.


[cid:CE9786CA-AE57-4B75-AF1F-8DE7CF0A525D][cid:1EB34C88-AD68-4165-A12E-C4EC3FB91D24][cid:43A6E371-267F-43F2-B161-91FB0F0D98A7]

However I wonder if it would be possible to change the readout ”hot water temperature” to another parameter instead  ”HOT_WATER_TOP_BT7”?

BR Alexander

26 apr. 2022 kl. 17:19 skrev Hubert Pruszyński ***@***.******@***.***>>:



@allindigital<https://github.com/allindigital> please update plugin to v0.0.13 and let me know if accessories from nibe works.

Currently supported:

  *   outdoor temperature
  *   hot water

**Comment by hpruszyn** [comment]
> However I wonder if it would be possible to change the readout ”hot water temperature” to another parameter instead  ”HOT_WATER_TOP_BT7”?

I think hot water current temperature is logically connected with activation of temporary lux - you know if you should start it and if it should be heating water. However I can add other simple accessory/service that will show hot water top temperature. And also I can add configuration to disable some accessories. Third option that came to my mind is to enable user to provide his own configuration file.

After longer thinking I will add last two enhancements to the roadmap.

**Comment by hpruszyn** [comment]
Proposed features moved to issues: https://github.com/hp-net/homebridge-nibe/issues/23 https://github.com/hp-net/homebridge-nibe/issues/22

**Comment by allindigital** [comment]
That current hot water temperature is connected is activation of temporary lux is great. I have actually tested it and it works great!

Looking forward the enhancements to come.

Have a good day!

Ps. I bought you a coffee.




27 apr. 2022 kl. 09:46 skrev Hubert Pruszyński ***@***.******@***.***>>:



However I wonder if it would be possible to change the readout ”hot water temperature” to another parameter instead ”HOT_WATER_TOP_BT7”?

I think hot water current temperature is logically connected with activation of temporary lux - you know if you should start it and if it should be heating water. However I can add other simple accessory/service that will show hot water top temperature. And also I can add configuration to disable some accessories. Third option that came to my mind is to enable user to provide his own configuration file.

After longer thinking I will add last two enhancements to roadmap.

**Comment by hpruszyn** [comment]
Thanks for the ☕  and stay tuned.

**Comment by allindigital** [comment]
:-)

27 apr. 2022 kl. 15:56 skrev Hubert Pruszyński ***@***.******@***.***>>:



Closed #14<https://github.com/hp-net/homebridge-nibe/issues/14>.
